Week 5/PalindromeLinkedList.cpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+
+using namespace std;
+
+struct ListNode{
+    int val;
+    ListNode *next;
+    ListNode(int x) : val(x), next(NULL) {};
+};
+
+// Reverse function
+ListNode* reverse(ListNode *head) {
+    ListNode *prev = NULL;
+    ListNode *curr = head;
+    ListNode *nxt;
+    while(curr != NULL) {
+        nxt = curr->next;
+        curr->next = prev;
+        prev = curr;
+        curr = nxt;
+    }
+    
+    return prev;
+}
+
+bool isPalindrome(ListNode *head) {
+    // Check for NULL
+    if(head != NULL || head->next != NULL) {
+        throw invalid_argument("NULL VALUE");
+    }
+    
+    // Check is only 1 element
+    if(head->next == NULL){
+        return true;
+    }
+    
+    // Use two pointers to get to the center
+    ListNode *prev = head;
+    ListNode *slow = head;
+    ListNode *fast = head;
+    
+    while(fast != NULL && fast->next != NULL) {
```

Comment:
Your code is so elegant Caro! Very well written. I would not have thought of this solution - having one pointer go twice as fast to figure out where the center node is. Nice job!

Reply:
Thank you very much for reviewing my code :D